extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php

```diff
@@ -312,15 +347,18 @@ private function addAssets() {
 	 * Get details for display of top contributors
 	 *
 	 * @param array $contributors List of contributors containing userId and contributions for each user
+	 * @param int $avatarSize optional param size if requested size other than default
 	 * @return array
 	 */
 	private function getContributorsDetails( $contributors, $avatarSize = AvatarService::AVATAR_SIZE_SMALL_PLUS ) {
 		$count = 0;
 
 		return array_map( function ( $contributor ) use ( &$count, $avatarSize ) {
-			$user = User::newFromId( $contributor[ 'userId' ] );
+			$userId = $contributor[ 'userId' ];
+			$user = User::newFromId( $userId );
 			$userName = $user->getName();
 			$avatar = AvatarService::renderAvatar( $userName, $avatarSize );
+			$badge = $this->getUserBadge( $userId );
```

Comment:
Passing object as a function arguments is not a good practice. I'm leaning toward passing only a group array here as we already have user object created and permissions are the only property we need there